File: src/revit_ifc_processor/get_ifc_model_face.py
import logging
import inspect
from Autodesk.Revit.DB import *

logger = logging.getLogger(__name__)


def get_ifc_model_face(element):
    """
    Getting geometrical face and store it into object
    So we can place Revit Family on it(on face instance).
    """
    function_name = inspect.currentframe().f_code.co_name   # testing Get current function name
    logging.info('Trying to get IFC model geometry face...')

    try:
        options = Options()
        options.ComputeReferences = True
        options.IncludeNonVisibleObjects = False

        geometry_element = element.Geometry[options]
        for geometry_instances in geometry_element:
            geometry_instances = geometry_instances.GetInstanceGeometry()
        for solid in geometry_instances:
            get_type = solid.GetType()      # Autodesk.Revit.DB.Solid
        # for faces in solid.Faces:
        #     print('element is: {}'.format(faces))  # Works => returns FACES
        faces_list = []
        for faces in solid.Faces:
            faces_list.append(faces)
        print('Faces list: {}'.format(faces_list))

        avg_face_area = sum(faces.Area for faces in solid.Faces) / len(faces_list)      # 9.4471253757322255 WORKS! calculate average face area
        biggest_face = []
        for big_face in faces_list:
            if big_face.Area >= avg_face_area:
                biggest_face.append(big_face)
        biggest_face_to_list = list(biggest_face)
        ifc_model_face = biggest_face_to_list[1]
        print('Biggest face: {}'.format(ifc_model_face))
        print('Biggest face: {}'.format(ifc_model_face.Area))

        print('IFC model face: {}'.format(ifc_model_face))
        return ifc_model_face

    except Exception as e:
        print('Error in get_ifc_model_face: {}'.format(e))
        logging.error('Error: {}, \n Function: {}'.format(e, function_name))
        return None

File: C#/src/Get_IFC_Model_Face.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Autodesk.Revit.DB;
using Autodesk.Revit.UI;

namespace revitapi_python_code_examples
{
    public class Get_ifc_model_face
    {

        public List<Face> Get_IFC_Model_Face(Element element)
        {
            try
            {
                Options options = new Options();
                options.ComputeReferences = true;
                options.IncludeNonVisibleObjects = false;

                GeometryElement geometry_element = element.get_Geometry(options);
                GeometryElement geometry_instance = null;
                foreach (var geometry_instances in geometry_element)
                {
                    if (geometry_instances is GeometryInstance)
                    {
                        GeometryInstance geometryInstance = geometry_instances as GeometryInstance;

                        GeometryElement instance_geometry = geometryInstance.GetInstanceGeometry();
                        geometry_instance = instance_geometry;
                    }
                }
                List<Face> faces_list = new List<Face>();
                foreach (Solid solid in geometry_instance)
                {
                    var get_type = solid.GetType();

                    foreach (Face face in solid.Faces)
                    {
                        faces_list.Add(face);
                    }
                }

                double avg_face_area = faces_list.Select(x => x.Area).Average();
                List<Face> biggest_face = new List<Face>();
                foreach (Face face in faces_list)
                {
                    if (face.Area > avg_face_area)
                    {
                        biggest_face.Add(face);
                    }
                }
                return biggest_face;
            }
            catch (Exception e)
            {
                TaskDialog.Show("Get_IFC_Model_Face", e.Message.ToString());
                return null;
            }

        }

    }
}